tmt/cli.py

```diff
@@ -870,3 +871,129 @@ def status(context, path, abandoned, active, finished, **kwargs):
+
+    Without any command, clean everything, stop the guests, remove
+    all runs and then remove all images. Search for runs in
+    /var/tmp/tmt, if runs are stored elsewhere, the path to them can
+    be set using a subcommand (either runs or guests subcommand).
+    """
+    clean_obj = tmt.Clean(parent=context.obj, context=context)
+    context.obj.clean = clean_obj
+    exit_code = 0
+    if context.invoked_subcommand is None:
+        echo(style('tmt cleanup', fg='red'))
+        # Set path to default
+        context.params['path'] = tmt.utils.WORKDIR_ROOT
+        # Create another level to the hierarchy so that logging indent is
+        # consistent between the command and subcommands
+        clean_obj = tmt.Clean(parent=clean_obj, context=context)
+        if os.path.exists(tmt.utils.WORKDIR_ROOT):
+            if not clean_obj.guests():
+                exit_code = 1
+            if not clean_obj.runs():
+                exit_code = 1
+        else:
+            echo(style(f'Directory {tmt.utils.WORKDIR_ROOT} does not exist, '
+                       f'skipping guest and run cleanup.', fg='yellow'))
+        clean_obj.images()
+        raise SystemExit(exit_code)
+
+
+@clean.command()
+@click.pass_context
+@click.argument('path', default=tmt.utils.WORKDIR_ROOT)
+@click.option(
+    '-l', '--last', is_flag=True, help='Clean the workdir of the last run.')
+@click.option(
+    '-i', '--id', 'id_', metavar="ID", help=
+    'Run id (name or directory path) to clean workdir of.')
+@click.option(
+    '-k', '--keep', type=int, help=
+    'The number of latest workdirs to keep, clean the rest')
+@verbose_debug_quiet
+@dry
+def runs(context, path, last, id_, keep, **kwargs):
+    """
+    Clean workdirs of past runs.
+
+    Remove all runs in /var/tmp/tmt by default. Path to where runs
+    should be searched can be specified using PATH argument.
+    """
+    echo(style('tmt cleanup', fg='red'))
+    defined = [last is True, id_ is not None, keep is not None]
+    if defined.count(True) > 1:
+        raise tmt.utils.GeneralError(
+            "Options --last, --id and --keep cannot be used together.")
+    if keep is not None and keep < 0:
+        raise tmt.utils.GeneralError("--keep must not be a negative number")
+    if not os.path.exists(path):
+        raise tmt.utils.GeneralError(f"Path {path} doesn't exist.")
+    exit_code = 0
+    if not tmt.Clean(parent=context.obj.clean, context=context).runs():
+        exit_code = 1
+    raise SystemExit(exit_code)
+
+
+@clean.command()
+@click.pass_context
+@click.argument('path', default=tmt.utils.WORKDIR_ROOT)
+@click.option(
+    '-l', '--last', is_flag=True, help='Stop the guest of the last run.')
+@click.option(
+    '-i', '--id', 'id_', metavar="ID", help=
+    'Run id (name or directory path) to stop the guest of.')
+@click.option('--container', is_flag=True, help='Stop container guests.')
+@click.option('--virtual', is_flag=True, help='Stop virtual guests.')
```

Comment:
Brainstorm: Would it make sense to use `-h --how` as an option to provide the requested provision method instead of defining options for each method separatele? Could be a bit more extensible for the future? Not sure...

Reply:
I was just following our initial brainstorm with this one. But your proposal probably makes more sense and will be more flexible, I will change it :+1: